ketcher-autotests/tests/Macromolecule-editor/Macro-Micro-Switcher/macro-micro-switcher.spec.ts

```diff
@@ -0,0 +1,439 @@
+import {
+  turnOnMacromoleculesEditor,
+  turnOnMicromoleculesEditor,
+} from '@utils/macromolecules';
+import { Page, test } from '@playwright/test';
+import {
+  FILE_TEST_DATA,
+  TopPanelButton,
+  clickInTheMiddleOfTheScreen,
+  openFileAndAddToCanvas,
+  pressButton,
+  selectTopPanelButton,
+  takeEditorScreenshot,
+  takeMonomerLibraryScreenshot,
+  takePageScreenshot,
+  waitForLoad,
+  waitForPageInit,
+  waitForRender,
+} from '@utils';
+
+async function pasteFromClipboard(page: Page, fileFormats: string) {
+  await selectTopPanelButton(TopPanelButton.Open, page);
+  await page.getByText('Paste from clipboard').click();
+  await page.getByRole('dialog').getByRole('textbox').fill(fileFormats);
+  await waitForLoad(page, async () => {
+    await pressButton(page, 'Add to Canvas');
+  });
+}
+
+async function addToFavoritesMonomers(page: Page) {
+  await page.getByTestId('Bal___beta-Alanine').getByText('★').click();
+  await page
+    .getByTestId('Phe4Me___p-Methylphenylalanine')
+    .getByText('★')
+    .click();
+  await page.getByTestId('meM___N-Methyl-Methionine').getByText('★').click();
+  await page.getByTestId('RNA-TAB').click();
+  await page.getByTestId('summary-Sugars').click();
+  await page.getByTestId('25R___2,5-Ribose').getByText('★').click();
+  await page.getByTestId('summary-Bases').click();
+  await page.getByTestId('baA___N-benzyl-adenine').getByText('★').click();
+  await page.getByTestId('summary-Phosphates').click();
+  await page.getByTestId('bP___Boranophosphate').getByText('★').click();
+  await page.getByTestId('CHEM-TAB').click();
+  await page.getByTestId('Test-6-Ch___Test-6-AP-Chem').getByText('★').click();
+}
+
+test.describe('Macro-Micro-Switcher', () => {
+  test.beforeEach(async ({ page }) => {
+    await waitForPageInit(page);
+    await turnOnMacromoleculesEditor(page);
```

Comment:
check for successful switch is redundant?

Reply:
Probably now this is really redundant because we don’t implement 'waitFor'. But in the future I think this will be relevant since we use the same thing in tests in micro mode.